elastic/kibana
# Restrict ability to disable features within a Space

Resolves https://github.com/elastic/kibana/issues/159392

## Summary

This PR hides the feature visibility section on the space management screen and disables adding `disabledFeatures` when creating or updating spaces using the REST API or spaces client on serverless.

## Screenshot

![Spaces-Elastic (2)](https://github.com/elastic/kibana/assets/190132/14d4900b-989d-420c-bddf-5ff70d305934)

## Testing

1. Start Kibana in serverless mode: `yarn start --serverless`
2. Edit default space and observe that the feature visibility section is not rendered
3. Quit Kibana and restart using classic mode: `yarn start`
4. Edit default space and observe that the feature visibility section is rendered correctly
5. Other considerations:
    - Disabling feature visibility in the classic offering should throw an error (`xpack.spaces.allowFeatureVisibility: false`)
    - Enabling feature visibility on serverless should throw an error (`xpack.spaces.allowFeatureVisibility: true`)
